bin/cpumask.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python3
+
+'''Convert CPU list(s) to CPU masks'''
+
+import argparse
+from dataclasses import dataclass
+
+
+@dataclass
+class global_vars:
+    '''Global variables'''
+
+    args = None
+
+
+def process_options():
+    '''Define the CLI argument parsing options'''
+
+    parser = argparse.ArgumentParser(description="CPU mask creator")
+
+    parser.add_argument("--cpus",
+                        dest = "cpus",
+                        help = "Use one or more times to define CPU and/or CPU lists to include in the mask",
+                        action = "append",
+                        type = str)
+    
+    myglobal.args = parser.parse_args()
+
+
+def main():
+    process_options()
+
+    cpus = set()
+    
+    for cpu1 in myglobal.args.cpus:
+        for cpu2 in cpu1.split(","):
+            cpu3 = cpu2.split("-")
+            if len(cpu3) == 1:
+                cpus.add(int(cpu3[0]))
+            else:
+                for cpu4 in range(int(cpu3[0]), int(cpu3[1])+1):
+                    cpus.add(cpu4)
```

Comment:
would need more error handling to make sure cpus are numeric ranges, single or comma-separated
```
ython3 cpumask.py --cpu 1,2-a,4-6
Traceback (most recent call last):
  File "cpumask.py", line 56, in <module>
    exit(main())
  File "cpumask.py", line 41, in main
    for cpu4 in range(int(cpu3[0]), int(cpu3[1])+1):
ValueError: invalid literal for int() with base 10: 'a'
```

Reply:
does it give an error for an invalid integer value?

Reply:
yes, but a traceback is not user friendly